### PR TITLE
gui plugin: project loading fixes

### PR DESCRIPTION
**Pull request by braunms**
see #539

### CODE REVIEW

File: plugins/gui/src/GUIWindows.cpp [hunk 1]
```diff
@@ -94,394 +94,411 @@ bool GUIWindows::CreateContext_GL(megamol::core::CoreInstance* instance) {
...
-        this->state.shutdown_triggered = true;
-        return false;
-    }
+    this->state.gui_post_drawing_enabled = this->state.gui_pre_drawing_enabled;
+    if (!this->state.gui_pre_drawing_enabled) {
+        // Reset ImGui context in core instance
+        if (this->core_instance != nullptr) {
+            this->core_instance->SetCurrentImGuiContext(nullptr);
+        }
+    } else {
```
Comment (geringsj) [inline plugins/gui/src/GUIWindows.cpp]: I suggest to do an early exit if drawing the GUI is disabled via `gui_pre_drawing_enabled` and `gui_post_drawing_enabled`. This leads to overall healthier code indentation and helps readability.

```
if(!drawing_enabled) {
    // do non-drawing stuff
    return;
}

// draw GUI
```

File: plugins/gui/src/GUIWindows.cpp [hunk 2]
```diff
@@ -722,221 +739,225 @@ bool megamol::gui::GUIWindows::ConsumeTriggeredScreenshot(void) {
...
 
-    // 1) Load all known calls from core instance ONCE ---------------------------
-    if (!this->configurator.GetGraphCollection().LoadCallStock(core_instance)) {
-        megamol::core::utility::log::Log::DefaultLog.WriteError(
-            "[GUI] Failed to load call stock once. [%s, %s, line %d]\n", __FILE__, __FUNCTION__, __LINE__);
-        return false;
-    }
-    /// TODO Load all known modules from core instance ONCE
-    /// XXX Omitted since this task takes ~2 seconds and would always block megamol for this period at start up!
+    if (this->state.gui_pre_drawing_enabled) {
```
Comment (geringsj) [inline plugins/gui/src/GUIWindows.cpp]: Suggesting early exit here and with the other `if (drawing_enabled)` conditionals, too.
```
if (! drawing_enabled) {
    return;
}

// draw GUI
```

Comment (braunms) [reply plugins/gui/src/GUIWindows.cpp]: Thanks, see commit 4720be1

File: plugins/gui/src/GUIWindows.cpp (hunk 1 above)
Comment (braunms) [reply plugins/gui/src/GUIWindows.cpp]: Thanks, see commit 4720be1

**Review by reinago — DISMISSED** [review]

**Review by geringsj — APPROVED** [review]

**Pull request merged**